src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
+        )
+
+    Config(; args...) = update_config!(new(), merge(defaults, Dict(args)))
+end
+config(; cargs...) = PreConfig(; cargs...)
+
+type PreConfig <: ConfigNode
+    cargs :: Dict{Symbol, Any}
+
+    PreConfig(; cargs...) = new(Dict(cargs))
+end
```

Comment:
The 2 steps I think are necessary if one wants to inherit config from the `document() root node` down to the final `page() "# Header" content`. Probably you are smarter so who knows - I could not figure it out in one step in a cleaner way. 

>just store everything in the Dict...

I was thinking also first to store all in the Dict even tried a version but it seems to me that it does not allow such nice controll of what is allowed and what is an error like types.